### PR TITLE
NMS-12407: Updated JCifs version

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-12407

Updated the version of JCIFS. Poller timeouts are now applied in JCifs.

### CODE REVIEW

File: protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java [hunk 1]
```diff
@@ -216,6 +223,23 @@ public boolean accept(SmbFile smbFile, String s) throws SmbException {
...
+        finalStaticField.setAccessible(true);
+
+        final Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(finalStaticField, finalStaticField.getModifiers() & ~Modifier.FINAL);
+
+        finalStaticField.set(null, value);
+    }
+
+    private void setJCifsTimeouts(int t) throws NoSuchFieldException, ClassNotFoundException, IllegalAccessException {
```
Comment (j-white) [inline protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java]: This doesn't appear to be thread safe. If we have two monitors with different timeouts executing at the same time, which timeout will they get?

Comment (j-white) [reply protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java]: There doesn't appear to be an easy way of working around this given the way JCifs is structured. Let's just add a note about this limitation in the docs and in the code.

File: protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java [hunk 2]
```diff
@@ -147,6 +148,12 @@ public boolean accept(SmbFile smbFile, String s) throws SmbException {
         // Initializing TimeoutTracker with default values
         TimeoutTracker tracker = new TimeoutTracker(parameters, DEFAULT_RETRY, DEFAULT_TIMEOUT);
 
+        try {
+            setJCifsTimeouts(tracker.getConnectionTimeout());
+        } catch (NoSuchFieldException | ClassNotFoundException | IllegalAccessException e) {
+            logger.error("Error setting JCifs timeouts ", e);
+        }
+
```
Comment (mvrueden) [inline protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java]: Can't we just put the error handling in the `setJCifsTimeouts(...)` method instead? Makes it much cleaner.

File: protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java [hunk 3]
```diff
@@ -216,6 +223,32 @@ public boolean accept(SmbFile smbFile, String s) throws SmbException {
         return serviceStatus;
     }
 
+    private void setFinalStatic(final Field finalStaticField, final Object value) throws NoSuchFieldException, IllegalAccessException {
```
Comment (mvrueden) [inline protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java]: Can we name this something like `setFieldValue(...)` or something, as this works for all fields, not only static ones.

File: protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java [hunk 4]
```diff
@@ -216,6 +223,32 @@ public boolean accept(SmbFile smbFile, String s) throws SmbException {
...
+     * @param t the timeout to be set
+     * @throws NoSuchFieldException
+     * @throws ClassNotFoundException
+     * @throws IllegalAccessException
+     */
+    private void setJCifsTimeouts(int t) throws NoSuchFieldException, ClassNotFoundException, IllegalAccessException {
+        final Class<?> clazz = Class.forName("jcifs.smb.SmbConstants");
+        setFinalStatic(clazz.getField("SO_TIMEOUT"), t);
+        setFinalStatic(clazz.getField("CONN_TIMEOUT"), t);
+        setFinalStatic(clazz.getField("DEFAULT_RESPONSE_TIMEOUT"), t + 5000);
```
Comment (mvrueden) [inline protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java]: Can we add a comment, why we add `5000` to it? Also `t` should be renamed to `timeoutInSeconds` or `timeoutInMilliseconds`, depending on what unit it is representing.

File: protocols/cifs/src/test/java/org/opennms/netmgt/poller/monitors/TimeoutTest.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        long startTime = System.currentTimeMillis();
+        final PollStatus pollStatus = jCifsMonitor.poll(svc, m);
+        long delta = System.currentTimeMillis() - startTime;
+        assertEquals(PollStatus.down(), pollStatus);
+
+        LOG.info("Checking " + delta + " <= " + limit);
+        assertTrue("Limit reached " + delta + " > " + limit, delta <= limit);
+    }
+
+
```
Comment (mvrueden) [inline protocols/cifs/src/test/java/org/opennms/netmgt/poller/monitors/TimeoutTest.java]: Nit: Unnecessary new line

File: protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java (hunk 2 above)
Comment (christianpape) [reply protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java]: done

File: protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java (hunk 3 above)
Comment (christianpape) [reply protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java]: done

File: protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java (hunk 4 above)
Comment (christianpape) [reply protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java]: done

File: protocols/cifs/src/test/java/org/opennms/netmgt/poller/monitors/TimeoutTest.java (hunk 5 above)
Comment (christianpape) [reply protocols/cifs/src/test/java/org/opennms/netmgt/poller/monitors/TimeoutTest.java]: done

**Pull request merged**